build/charts/antrea/crds/packetcapture.yaml

```diff
@@ -0,0 +1,193 @@
+          name: Destination-Pod
+          type: string
+          priority: 10
+        - jsonPath: .spec.destination.ip
+          description: The IP address of the destination.
+          name: Destination-IP
+          type: string
+          priority: 10
+        - jsonPath: .spec.timeout
+          description: Timeout in seconds.
+          name: Timeout
+          type: integer
+          priority: 10
+        - jsonPath: .metadata.creationTimestamp
+          name: Age
+          type: date
+      schema:
+        openAPIV3Schema:
+          type: object
+          required:
+            - spec
+          properties:
+            spec:
+              type: object
+              required:
+                - source
+                - captureConfig
+                - destination
+              anyOf:
+                - properties:
+                    source:
+                      required: [pod]
+                - properties:
+                    destination:
+                      required: [pod]
+              properties:
+                source:
+                  type: object
+                  oneOf:
+                    - required:
+                      - pod
+                    - required:
+                      - ip
+                  properties:
+                    pod:
+                      type: object
+                      properties:
+                        namespace:
+                          type: string
+                        name:
+                          type: string
+                    ip:
+                      type: string
+                      oneOf:
+                        - format: ipv4
+                        - format: ipv6
+                destination:
+                  type: object
+                  oneOf:
+                    - required:
+                      - pod
+                    - required:
+                      - ip
+                    - required:
+                      - service
+                  properties:
+                    pod:
+                      type: object
+                      properties:
+                        namespace:
+                          type: string
+                        name:
+                          type: string
+                    service:
+                      type: object
+                      properties:
+                        namespace:
+                          type: string
+                        name:
+                          type: string
+                    ip:
+                      type: string
+                      oneOf:
+                        - format: ipv4
+                        - format: ipv6
+                packet:
+                  type: object
+                  properties:
+                    ipFamily:
+                      type: string
+                      enum: [IPv4, IPv6]
+                      default: IPv4
+                    protocol:
+                      x-kubernetes-int-or-string: true
```

Comment:
This should be limited to three supported types: udp/tcp/icmp

Reply:
https://github.com/antrea-io/antrea/pull/6257#discussion_r1772285151

there is a discussion about this before. I personally think maybe SCTP is not needed for now.

Reply:
@luolanzone there is no reason to limit protocols here in my opinion. We are capturing traffic, not generating traffic, and Pod traffic can use any protocol. Only Services are limited to TCP / UDP / SCTP.
We could restrict protocol strings to what we "know" (as we need to do the mapping), but protocol numbers should be arbitrary.
If there is a way to restrict the string value to known values, while allowing arbitrary numeric values, then we can go for it.

Reply:
In the code MR we are planning to remove the support for service. As we don't have a good way to distinguish the traffic between service and normal pod -> pod. I think we can keep the current format, and at least add some tests for SCTP.  Using  tcpdump filter to perform the capture(we are switching to gopacket/libpcap from ovs), it should work on any protocol value.

Reply:
I don't think that's the point I was trying to make.
I was trying to convey that K8s itself doesn't limit what transport protocols can be used for Pod-to-Pod traffic, and so this feature shouldn't restrict it either by forcing the protocol to be one of TCP, UDP, ICMP, SCTP. This is why we should support an arbitrary protocol number here, even though we should also support some pre-defined strings for convenience.
There is no need to test SCTP support specifically. From an Antrea perspective, there is no real difference in the implementation between protocols. It's just a number that we use to define a packet capture filter.

Reply:
sure !